pkg/rke2/serviceaccount.go

```diff
@@ -0,0 +1,64 @@
+package rke2
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func restrictServiceAccount(ctx context.Context, namespace string, cs *kubernetes.Clientset) error {
+	var sa *v1.ServiceAccount
+	var err error
```

Comment:
Can you scope lock the err value on 18?

Reply:
I have scope locked the err and for loop.